pkg/virtctl/vm/expand_test.go

```diff
@@ -0,0 +1,172 @@
+	vmName := "testvm"
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		expandSpecInterface = kubecli.NewMockExpandSpecInterface(ctrl)
+		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+		vm = kubecli.NewMinimalVM(vmName)
+	})
+
+	It("should fail expand", func() {
+		cmd := clientcmd.NewRepeatableVirtctlCommand("expand")
+		Expect(cmd()).NotTo(Succeed())
+	})
+
+	Context("called on VM", func() {
+		It("should succeed when called on vm", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(context.Background(), vmName)
+
+			cmd := clientcmd.NewVirtctlCommand("expand", "--vm", vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		})
+
+		DescribeTable("should succeed when called with ", func(formatName string) {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(context.Background(), vmName)
+
+			cmd := clientcmd.NewVirtctlCommand("expand", outputFormat, formatName, "--vm", vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		},
+			Entry("supported format json", "json"),
+			Entry("supported format yaml", "yaml"),
+		)
+
+		It("should fail when called on non existing vm", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(context.Background(), nonExistingVM).Return(nil, fmt.Errorf("\"%s\" not found", nonExistingVM))
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", "--vm", nonExistingVM)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).Should(MatchError("error expanding VirtualMachine - non-existing-vm in namespace - default: \"non-existing-vm\" not found"))
+		})
+
+		It("should fail when called with non supported output format", func() {
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", outputFormat, invalidFormat, "--vm", vmName)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).Should(MatchError("error not supported output format defined: test-format"))
+		})
+	})
+
+	Context("with file input", func() {
+		const (
+			vmSpec = `apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: testvm
+spec:
+  runStrategy: Always
+  template:
+    spec:
+      domain:
+        devices: {}
+        machine:
+          type: q35
+        resources: {}
+        volumes:
+status:
+`
+			invalidYaml = `apiVersion: kubevirt.io/v1kind: VirtualMachine`
+		)
+		BeforeEach(func() {
+			file, err = os.CreateTemp("", "file-*")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should succeed when called with existing file input", func() {
```

Comment:
Currently, this test description will say `with file input should succeed when called with existing file input`, which is a little redundant and I feel that "existing file input" is a vague description. 

It might be worth considering changing this to `"should succeed when called with valid file input"` for clarity and a more accurate test description.